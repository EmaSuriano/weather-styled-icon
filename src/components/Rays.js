import styled from 'styled-components';
import { ThemePropType, defaultTheme } from '../constants';

const Rays = styled.div.attrs({
  'data-e2e': 'rays',
})`
  position: absolute;
  top: -2em;
  left: 50%;
  display: block;
  width: 0.375em;
  height: 1.125em;
  margin-left: -0.1875em;
  border-radius: 0.25em;
  background: ${({ theme }) => theme.raysColor};
  box-shadow: 0 5.375em ${({ theme }) => theme.raysColor};

  &:before,
  &:after {
    content: '';
    position: absolute;
    top: 0em;
    left: 0em;
    display: block;
    width: 0.375em;
    height: 1.125em;
    transform: rotate(60deg);
    transform-origin: 50% 3.25em;
    border-radius: 0.25em;
    background: ${({ theme }) => theme.raysColor};
    box-shadow: 0 5.375em ${({ theme }) => theme.raysColor};
  }

  &:before {
    transform: rotate(120deg);
  }
`;

Rays.propTypes = {
  theme: ThemePropType,
};

Rays.defaultProps = {
  theme: defaultTheme,
};

Rays.displayName = 'Rays';

export default Rays;
