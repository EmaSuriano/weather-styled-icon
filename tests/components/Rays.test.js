import React from 'react';
import Rays from '../../src/components/Rays';
import { shallow } from 'enzyme';
import 'jest-styled-components';

describe('<Rays />', () => {
  it('should should render div with theme', () => {
    const wrapper = shallow(<Rays theme={{ raysColor: 'yellow' }} />);

    expect(wrapper).toMatchSnapshot();
  });
});
