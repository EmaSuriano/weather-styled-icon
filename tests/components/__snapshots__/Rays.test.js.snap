// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Rays /> should should render div with theme 1`] = `
.c0 {
  position: absolute;
  top: -2em;
  left: 50%;
  display: block;
  width: 0.375em;
  height: 1.125em;
  margin-left: -0.1875em;
  border-radius: 0.25em;
  background: yellow;
  box-shadow: 0 5.375em yellow;
}

.c0:before,
.c0:after {
  content: '';
  position: absolute;
  top: 0em;
  left: 0em;
  display: block;
  width: 0.375em;
  height: 1.125em;
  -webkit-transform: rotate(60deg);
  -ms-transform: rotate(60deg);
  transform: rotate(60deg);
  -webkit-transform-origin: 50% 3.25em;
  -ms-transform-origin: 50% 3.25em;
  transform-origin: 50% 3.25em;
  border-radius: 0.25em;
  background: yellow;
  box-shadow: 0 5.375em yellow;
}

.c0:before {
  -webkit-transform: rotate(120deg);
  -ms-transform: rotate(120deg);
  transform: rotate(120deg);
}

<div
  className="c0"
/>
`;
